lib/middleman-images/image.rb

```diff
@@ -34,6 +34,14 @@ def resize(image_path, options)
           raise 'The gem "mini_magick" is required for image resizing. Please install "mini_magick" or remove the resize option.'
         end
 
+        image_ext = File.extname(image_path)
+        case image_ext
+        when ".svg"
+          app.logger.warn "WARNING: You are trying to resize #{image_path}. Resizing SVG files will lead to ImageMagick creating an SVG with an embedded binary image thus making the file way bigger."
+        when ".gif"
+          app.logger.warn "WARNING: You are trying to resize #{image_path}. Resizing GIF files will lead to ImageMagick removing the animation."
+        end
```

Comment:
As was discussed and mentioned in #33: it would be best to warn about an unwanted resize option *and* ignore the resize option in those cases. Most users might not notice the warning directly and will receive receive badly resized images as a result.

Reply:
For that goal a hardcoded hash list containing warnings and to-be-ignored extensions could help to reduce code duplication.